src/lamplib/tasks/download.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import urllib.request
+
+from context import Context
+
+class Downloader():
+    """
+    Abstract base class for things that can be downloaded.
+
+    :return: The directory the target was installed to.
+    """
+
+    def __init__(self, os_family, distro, install_dir, name):
+        self._os_family = os_family
+        self._distro = distro
+        self._name = name
+
+        # Install dir is the directory we create the actual result in.
+        self._install_dir = install_dir
+        self.result_dir = None
+
+    def fetch_and_install(self):
+        """
+        Download the target and install to its directory.
+
+        :return: Whether the operation succeeded.
+        """
+        if not os.path.exists(self._install_dir):
+            logging.critical(
+                'Please create the parent directory for %s: '
+                '`sudo mkdir -p %s; sudo chown $USER %s`', self._name, self._install_dir, self._install_dir)
+            return False
```

Comment:
let's (at least try to) quote this properly for the user
`sudo mkdir -p "%s"; sudo chown "$USER" "%s"`